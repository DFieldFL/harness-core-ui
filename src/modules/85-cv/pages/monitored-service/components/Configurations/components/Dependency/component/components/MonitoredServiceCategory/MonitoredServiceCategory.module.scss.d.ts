/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly monitoredServiceCategory: string
  readonly categoryName: string
  readonly application: string
  readonly infrastructure: string
  readonly monitoredServiceCategoryVertical: string
}
export default styles
