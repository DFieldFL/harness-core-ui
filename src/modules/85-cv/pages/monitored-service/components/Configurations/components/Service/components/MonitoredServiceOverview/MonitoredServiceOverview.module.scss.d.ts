/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly monitoredService: string
  readonly nameTagsDescription: string
  readonly divider: string
  readonly dropdown: string
  readonly disabled: string
}
export default styles
