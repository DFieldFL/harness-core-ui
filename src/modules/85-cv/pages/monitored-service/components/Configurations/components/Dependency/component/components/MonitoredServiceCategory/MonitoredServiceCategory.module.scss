.monitoredServiceCategory {
  display: flex;
  justify-content: center;
  padding: 0 var(--spacing-xsmall) !important;
  height: 25px;
  box-sizing: border-box;
  border-radius: 4px;

  .categoryName {
    font-size: var(--font-size-xsmall) !important;
    text-transform: uppercase;
    font-weight: bolder !important;
    letter-spacing: 2px !important;
  }

  &.application {
    background: #effbff !important;
    border: 1px solid #cdf4fe;

    .categoryName {
      color: var(--primary-8) !important;
    }
  }

  &.infrastructure {
    background: var(--purple-50) !important;
    border: 1px solid var(--purple-100);

    .categoryName {
      color: var(--purple-500) !important;
    }
  }
}

.monitoredServiceCategoryVertical {
  height: 100%;
  align-items: flex-end;
  writing-mode: vertical-rl;
  transform: rotate(180deg);
  position: absolute;
  top: 0;
  left: -5px;

  &.application {
    .categoryName {
      align-items: flex-end !important;
    }
  }

  &.infrastructure {
    .categoryName {
      align-items: flex-end !important;
    }
  }
}
