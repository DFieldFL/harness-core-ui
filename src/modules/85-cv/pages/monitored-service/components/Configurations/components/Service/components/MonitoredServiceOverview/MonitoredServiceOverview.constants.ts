import type { SelectOption } from '@wings-software/uicore'

export const MonitoredServiceType = {
  INFRASTRUCTURE: 'Infrastructure',
  APPLICATION: 'Application'
}

export const MonitoredServiceTypeOptions: SelectOption[] = [
  { label: MonitoredServiceType.APPLICATION, value: MonitoredServiceType.APPLICATION },
  { label: MonitoredServiceType.INFRASTRUCTURE, value: MonitoredServiceType.INFRASTRUCTURE }
]
