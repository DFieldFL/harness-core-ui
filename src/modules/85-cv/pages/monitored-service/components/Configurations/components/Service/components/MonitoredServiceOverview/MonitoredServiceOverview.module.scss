.monitoredService {
  margin-top: var(--spacing-none) !important;
  .nameTagsDescription {
    width: 400px !important;
  }
  .divider {
    border: 0;
    border-top: 1px solid var(--grey-100);
    margin: var(--spacing-xlarge) 0 !important;
  }
}

.dropdown {
  width: 400px !important;
}

.disabled {
  pointer-events: none;
  cursor: not-allowed;
  opacity: 0.7;
  input {
    background-color: var(--grey-100);
    color: var(--grey-400);
  }
}
