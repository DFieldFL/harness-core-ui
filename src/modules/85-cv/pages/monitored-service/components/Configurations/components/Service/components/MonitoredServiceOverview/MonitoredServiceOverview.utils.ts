import type { FormikProps } from 'formik'
import type { SelectOption } from '@wings-software/uicore'

export function generateMonitoredServiceName(serviceIdentifier?: string, envIdentifier?: string): string {
  let name = ''
  if (serviceIdentifier?.length) {
    name += serviceIdentifier
  }
  if (envIdentifier?.length) {
    name += name?.length ? `_${envIdentifier}` : envIdentifier
  }

  return name
}

export function updatedMonitoredServiceNameForEnv(formik: FormikProps<any>, environment?: SelectOption): void {
  const { values } = formik || {}
  const monitoredServiceName = generateMonitoredServiceName(values.serviceRef, environment?.value as string)
  formik.setValues({
    ...values,
    environmentRef: environment?.value,
    name: monitoredServiceName,
    identifier: monitoredServiceName
  })
}

export function updateMonitoredServiceNameForService(formik: FormikProps<any>, service?: SelectOption): void {
  const { values } = formik || {}
  const monitoredServiceName = generateMonitoredServiceName(service?.value as string, values.environmentRef)
  formik.setValues({
    ...values,
    serviceRef: service?.value,
    name: monitoredServiceName,
    identifier: monitoredServiceName
  })
}
