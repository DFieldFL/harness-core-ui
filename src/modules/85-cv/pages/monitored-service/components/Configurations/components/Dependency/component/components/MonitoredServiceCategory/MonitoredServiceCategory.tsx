import React from 'react'
import cx from 'classnames'
import { Text, Color, Container } from '@wings-software/uicore'
import type { MonitoredServiceDTO } from 'services/cv'
import { MonitoredServiceType } from '@cv/pages/monitored-service/components/Configurations/components/Service/components/MonitoredServiceOverview/MonitoredServiceOverview.constants'
import css from './MonitoredServiceCategory.module.scss'

export default function MonitoredServiceCategory({ type }: { type?: MonitoredServiceDTO['type'] }): JSX.Element {
  switch (type) {
    case 'Infrastructure':
      return (
        <Container className={cx(css.monitoredServiceCategory, css.infrastructure)}>
          <Text className={css.categoryName} iconProps={{ size: 18, color: Color.PURPLE_500 }} icon="infrastructure">
            {MonitoredServiceType.INFRASTRUCTURE}
          </Text>
        </Container>
      )
    case 'Application':
      return (
        <Container className={cx(css.monitoredServiceCategory, css.application)}>
          <Text className={css.categoryName} iconProps={{ size: 13, color: Color.PRIMARY_8 }} icon="dashboard">
            {MonitoredServiceType.APPLICATION}
          </Text>
        </Container>
      )
    default:
      return <Container />
  }
}
