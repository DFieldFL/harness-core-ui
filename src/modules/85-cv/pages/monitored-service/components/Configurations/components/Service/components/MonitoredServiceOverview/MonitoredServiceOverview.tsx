import React from 'react'
import { Layout, FormInput } from '@wings-software/uicore'
import { NameIdDescriptionTags } from '@common/components'
import { useStrings } from 'framework/strings'
import CardWithOuterTitle from '@cv/pages/health-source/common/CardWithOuterTitle/CardWithOuterTitle'
import {
  useGetHarnessServices,
  useGetHarnessEnvironments,
  HarnessServiceAsFormField,
  HarnessEnvironmentAsFormField
} from '@cv/components/HarnessServiceAndEnvironment/HarnessServiceAndEnvironment'
import { MonitoredServiceTypeOptions } from './MonitoredServiceOverview.constants'
import {
  updateMonitoredServiceNameForService,
  updatedMonitoredServiceNameForEnv
} from './MonitoredServiceOverview.utils'
import type { MonitoredServiceOverviewProps } from './MonitoredSourceOverview.types'
import css from './MonitoredServiceOverview.module.scss'

export default function MonitoredServiceOverview(props: MonitoredServiceOverviewProps): JSX.Element {
  const { formikProps, isEdit } = props
  const { getString } = useStrings()
  const { serviceOptions, setServiceOptions } = useGetHarnessServices()
  const { environmentOptions, setEnvironmentOptions } = useGetHarnessEnvironments()
  const values = formikProps.values || {}

  return (
    <CardWithOuterTitle title={getString('overview')} className={css.monitoredService}>
      {!isEdit ? (
        <>
          <Layout.Horizontal spacing="large">
            <FormInput.Select name="type" items={MonitoredServiceTypeOptions} label={getString('typeLabel')} />
            <HarnessServiceAsFormField
              customRenderProps={{
                name: 'serviceRef',
                label: getString('cv.healthSource.serviceLabel')
              }}
              serviceProps={{
                className: css.dropdown,
                disabled: isEdit,
                item: serviceOptions.find(item => item?.value === values.serviceRef),
                options: serviceOptions,
                onSelect: selectedService => updateMonitoredServiceNameForService(formikProps, selectedService),
                onNewCreated: newOption => {
                  if (newOption?.identifier && newOption.name) {
                    const newServiceOption = { label: newOption.name, value: newOption.identifier }
                    setServiceOptions([newServiceOption, ...serviceOptions])
                    updateMonitoredServiceNameForService(formikProps, newServiceOption)
                  }
                }
              }}
            />
            <HarnessEnvironmentAsFormField
              customRenderProps={{
                name: 'environmentRef',
                label: getString('cv.healthSource.environmentLabel')
              }}
              environmentProps={{
                className: css.dropdown,
                disabled: isEdit,
                item: environmentOptions.find(item => item?.value === values.environmentRef),
                options: environmentOptions,
                onSelect: environment => updatedMonitoredServiceNameForEnv(formikProps, environment),
                onNewCreated: newOption => {
                  if (newOption?.identifier && newOption.name) {
                    const newEnvOption = { label: newOption.name, value: newOption.identifier }
                    setEnvironmentOptions([newEnvOption, ...environmentOptions])
                    updatedMonitoredServiceNameForEnv(formikProps, newEnvOption)
                  }
                }
              }}
            />
          </Layout.Horizontal>
          <hr className={css.divider} />
        </>
      ) : null}
      <NameIdDescriptionTags
        formikProps={formikProps}
        inputGroupProps={{ disabled: true }}
        className={css.nameTagsDescription}
        identifierProps={{
          isIdentifierEditable: false,
          inputLabel: getString('cv.monitoredServices.monitoredServiceName')
        }}
      />
    </CardWithOuterTitle>
  )
}
