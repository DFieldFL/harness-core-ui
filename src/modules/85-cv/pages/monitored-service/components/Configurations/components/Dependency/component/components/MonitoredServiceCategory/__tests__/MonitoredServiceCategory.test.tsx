import React from 'react'
import { render, waitFor } from '@testing-library/react'
import { MonitoredServiceType } from '@cv/pages/monitored-service/components/Configurations/components/Service/components/MonitoredServiceOverview/MonitoredServiceOverview.constants'
import MonitoredServiceCategory from '../MonitoredServiceCategory'

describe('Unit tests for MonitoredServiceCategory', () => {
  test('Ensure infra type is rendered correctly', async () => {
    const { container, getByText } = render(<MonitoredServiceCategory type="Infrastructure" />)
    await waitFor(() => expect(container.querySelector('[class*="infrastructure"]')).not.toBeNull())
    getByText(MonitoredServiceType.INFRASTRUCTURE)
    expect(container).toMatchSnapshot()
  })
  test('Ensure application typee is rendered correctly', async () => {
    const { container, getByText } = render(<MonitoredServiceCategory type="Application" />)
    await waitFor(() => expect(container.querySelector('[class*="application"]')).not.toBeNull())
    getByText(MonitoredServiceType.APPLICATION)
    expect(container).toMatchSnapshot()
  })
  test('Ensure null type is rendered correctly', async () => {
    const { container } = render(<MonitoredServiceCategory />)
    await waitFor(() => expect(container.querySelector('[class*="application"]')).toBeNull())
    expect(container.querySelector('[class*="infrastructure"]')).toBeNull()
    expect(container).toMatchSnapshot()
  })
})
