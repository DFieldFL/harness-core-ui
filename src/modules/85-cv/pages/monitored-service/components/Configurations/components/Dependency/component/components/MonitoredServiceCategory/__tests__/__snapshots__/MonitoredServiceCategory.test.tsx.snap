// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for MonitoredServiceCategory Ensure application typee is rendered correctly 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main monitoredServiceCategory application"
  >
    <p
      class="StyledProps--font StyledProps--main categoryName"
      style="display: flex; align-items: center;"
    >
      <span
        class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-primary8"
        data-icon="dashboard"
      >
        <svg
          height="13"
          viewBox="0 0 16 16"
          width="13"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M8.656 2H14v4H8.656zm0 12V7.344H14V14zM2 14v-4h5.344v4zm0-5.344V2h5.344v6.656z"
            fill="#fff"
            opacity="0.691"
          />
        </svg>
      </span>
      Application
    </p>
  </div>
</div>
`;

exports[`Unit tests for MonitoredServiceCategory Ensure infra type is rendered correctly 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main monitoredServiceCategory infrastructure"
  >
    <p
      class="StyledProps--font StyledProps--main categoryName"
      style="display: flex; align-items: center;"
    >
      <span
        class="bp3-icon StyledProps--main StyledProps--padding-right-xsmall StyledProps--color StyledProps--color-purple500"
        data-icon="infrastructure"
      >
        <svg
          fill="currentColor"
          height="18"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4h16v16H4z"
            opacity="0.01"
          />
          <path
            clip-rule="evenodd"
            d="M12.487 8.343a.2.2 0 00.08.17l6.287 4.692a1 1 0 01.006 1.598l-6.257 4.74a1 1 0 01-1.207.002l-6.325-4.773a1 1 0 01-.026-1.577l.664-.536a.2.2 0 01.248-.002l.207.161a.3.3 0 01-.006.479l-.593.436a.2.2 0 00-.002.32l5.618 4.253a.2.2 0 00.321-.159v-2.515L5.066 10.8a1 1 0 010-1.6l6.344-4.758a1 1 0 011.187-.01l6.408 4.646a1 1 0 01-.036 1.645l-.905.597-.482-.471a.2.2 0 01.037-.315l.747-.449a.2.2 0 00.011-.335l-5.471-3.825a.2.2 0 00-.315.155l-.104 2.263zm-.767-2.476a.2.2 0 00-.32-.16L5.972 9.775a.2.2 0 000 .32l5.753 4.346a.5.5 0 00.567.025l2.642-1.65a.2.2 0 01.233.014l.452.37a.2.2 0 01-.02.324l-3.217 2.02a.2.2 0 00-.093.17v2.468a.2.2 0 00.322.158l5.42-4.18a.2.2 0 00-.001-.318l-5.853-4.469a.2.2 0 00-.245.002l-2.438 1.92a.2.2 0 01-.258-.008l-.382-.346a.2.2 0 01.011-.305l2.78-2.186a.2.2 0 00.075-.158V5.867z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
      Infrastructure
    </p>
  </div>
</div>
`;

exports[`Unit tests for MonitoredServiceCategory Ensure null type is rendered correctly 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main"
  />
</div>
`;
